tools/test-in-lxd.sh

```diff
@@ -1,36 +1,26 @@
 #!/usr/bin/bash
-series=$1
+set -eux
 
-set -x
-
-build_out=$(./tools/build.sh $series)
-hash=$(echo $build_out | jq -r .state_hash)
-deb=$(echo $build_out | jq -r .debs[] | grep tools)
+VM=${VM:-0}
+series=${1:-jammy}
+build_out=$(./tools/build.sh "$series")
+hash=$(echo "$build_out" | jq -r .state_hash)
+deb=$(echo "$build_out" | jq -r '.debs[]' | grep tools)
 name=ua-$series-$hash
 
-if [ -n "$VM" ]; then
+flags=
+if [ "$VM" -ne 0 ]; then
     flags="--vm"
 fi
 
-lxc delete $name --force
-lxc launch ubuntu-daily:$series $name $flags
+lxc delete "$name" --force || true
+lxc launch "ubuntu-daily:${series}" "$name" $flags
 sleep 5
-if [ -n "$VM" ]; then
+if [[ "$VM" -ne 0 ]]; then
     echo "vms take a while before the agent is ready"
     sleep 30
 fi
-lxc file push $deb $name/tmp/ua.deb
-
-if [ -n "$SHELL_BEFORE" ]; then
```

Comment:
This is actually useful;
By setting the envvar when calling the script, we have a shell there with the base version of UA before the tested version is installed; sometimes we want to tweak the system or put it in a specific state before `dpkg -i` happens

Reply:
Reverted. I thought it was dead code.